integration_test/Bindings/Python/dialects/rtl.py

```diff
@@ -45,7 +45,45 @@ def top(a, b):
       a, b = swap(a, b)
       return a, b
 
-  m.operation.print()
+    one_input = rtl.RTLModuleOp(
+        name="one_input",
+        input_ports=[("a", i32)],
+        body_builder=lambda m: rtl.OutputOp([]),
+    )
+    two_inputs = rtl.RTLModuleOp(
+        name="two_inputs",
+        input_ports=[("a", i32), ("b", i32)],
+        body_builder=lambda m: rtl.OutputOp([]),
+    )
+    one_output = rtl.RTLModuleOp(
+        name="one_output",
+        output_ports=[("a", i32)],
+        body_builder=lambda m: rtl.OutputOp(
+            [rtl.ConstantOp(i32, IntegerAttr.get(i32, 46)).result]),
+    )
+
+    # CHECK-LABEL: rtl.module @instance_builder_tests
+    def instance_builder_body(module):
+      # CHECK: %[[INST1_RESULT:.+]] = rtl.instance "inst1" @one_output()
+      inst1 = one_output.create(module, "inst1")
+
+      # CHECK: rtl.instance "inst2" @one_input(%[[INST1_RESULT]])
+      inst2 = one_input.create(module, "inst2", {"a": inst1.get_port("a")})
```

Comment:
Let me think about it, but I think adding back __getattr__ is a good idea. That does mean that if `inst1.a` returns a `Value`, then `inst4.b.set` would mean we need to put that method on Value. I'm starting to lean towards attribute like getters and a `connect` free function, but again let's continue to iterate.

Reply:
I've added back `__getattr__` to get either input or output ports, and I think that should play nicely with whatever we do here. There are some considerations either way. I think if we do the `connect()` approach, it will need to be a method on the parent module, so we have access to its backedge builder. For now I will just use the explicit `set_input_port` call.